core/modules/catalog-next/store/category/actions.ts

```diff
@@ -140,7 +141,7 @@ const actions: ActionTree<CategoryState, RootState> = {
       categorySearchOptions.filters.id = searchedIds.filter(categoryId => !getters.getCategoriesMap[categoryId] && !getters.getNotFoundCategoryIds.includes(categoryId))
     }
     if (!searchingByIds || categorySearchOptions.filters.id.length) {
-      categorySearchOptions.filters = Object.assign(config.entities.category.filterFields, categorySearchOptions.filters)
+      categorySearchOptions.filters = Object.assign(cloneDeep(config.entities.category.filterFields), categorySearchOptions.filters ? cloneDeep(categorySearchOptions.filters) : {})
```

Comment:
```suggestion
      categorySearchOptions.filters = cloneDeep(Object.assign(config.entities.category.filterFields, categorySearchOptions.filters ? categorySearchOptions.filter) : {}))
```

Reply:
In this suggestion, the Object.assign() function would mutate the config, by merging categorySearchOptions.filters into config.entities.category.filterFields.
We need to either clone both config.entities.category.filterFields and categorySearchOptions, or we could use an empty object {} as the first parameter of Object.assign()

Reply:
yeah you are right, ignore it 👍